drivers/lps331ap/lps331ap_saul.c

```diff
@@ -32,8 +32,24 @@ static int read(const void *dev, phydat_t *res)
     return 1;
 }
 
-const saul_driver_t lps331ap_saul_driver = {
-    .read = read,
+static int read_temp(const void *dev, phydat_t *res)
+{
+    res->val[0] = (int16_t)(lps331ap_read_temp((const lps331ap_t *)dev) / 10);
+    memset(&(res->val[1]), 0, 2 * sizeof(int16_t));
```

Comment:
or more precisely: anything that uses SAUL to read sensors should only process that much of the `res` array